src/main/java/mate/academy/springboot/swagger/service/Impl/SortParserServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package mate.academy.springboot.swagger.service.Impl;
+
+import mate.academy.springboot.swagger.service.SortParserService;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class SortParserServiceImpl implements SortParserService {
+
+    @Override
+    public List<Sort.Order> orders(String sortBy) {
+        List<Sort.Order> orders = new ArrayList<>();
+        if (sortBy.contains(":")) {
+            String[] sortingFields = sortBy.split(";");
+            for (String field : sortingFields) {
+                Sort.Order order;
+                if (field.contains(":")) {
+                    String[] fieldsAndDirections = field.split(":");
+                    order = new Sort.Order(Sort.Direction.valueOf(fieldsAndDirections[1]),
+                            fieldsAndDirections[0]);
+                } else {
+                    order = new Sort.Order(Sort.Direction.DESC, field);
+                }
```

Comment:
Lot of contains() and split() with ";" and ":" (which I doubt is correct) with 3 tabulations to the right and calls of inner classes.
It's not readable at all.  
Please simplify: make constants, make separate methods with proper naming etc.